### ISSUE TITLE
Action to pause/resume key mapper

### ISSUE THREAD

**Issue by D-Kaminari** [keymapperorg/KeyMapper#868]
**Developer TODO (don't remove)**
- [ ] create new branch. put issue number at start of name
- [ ] update documentation
- [ ] exclude the key map when you pause key maps
- [ ] squash and merge
- [ ] delete branch

**Comment by sds100** [comment]
Hi, do you mean action?

**Comment by D-Kaminari** [comment]
Hi. Well, I meant for the thing itself to be a Constraint, but it would be
triggered by any Action. So if you were asking about the key press being an
Action, then yes, but if you were asking if I meant Action instead of
Constraint, then no, but I think that it doesn't matter whether you make it
an Action or a Constraint.
Also, why isn't your app on the Play Store any more? I had to factory reset
my device some time ago, and now when I search for it to reinstall it, it
isn't there any more. The only thing that's there is the GUI Keyboard app.
Please add it back.
Thanks.

**Comment by sds100** [comment]
Hmm, it should still be on the Play store. I have had some users report this and am not sure why it is missing. What device do you have (and Android version).

**Comment by sds100** [comment]
It sounds like you want an action to pause/resume Key Mapper. An action is triggered by a trigger (i.e key presses) and a constraint only allows an action to be triggered in certain situations.

**Comment by D-Kaminari** [comment]
Hi. Thank you for clarifying. Yes, I did mean it as an action.
Also, just to remind you, your app isn't on the Play Store any more. Could
you please let somebody know to fix it, or could you please provide me with
some sort of APK file of the app once you add the update?
Thanks.

**Comment by sds100** [comment]
> 

What device are you using?

**Comment by sds100** [comment]
@D-Kaminari can you download it again? I think i fixed the issue on Google Play.

**Comment by githubsterer** [comment]
Yeah, not sure how you deal with the resume app from a paused app thing, but you do have this already as a quick tile so it can clearly be done somehow. Maybe it needs to be offloaded to the secondary app? But the idea would be, when paired with something like touch the notch, or your folding button, to be able to quickly get to that control wherever you are working on the phone.

**Comment by githubsterer** [comment]
What is the specific activity you call with the quick settings to pause Key Mapper? I'm trying to see how I can activate it using Tasker.

**Comment by jambl3r** [comment]
It isn't an activity. There is an intent to pause and resume all key maps:
https://docs.keymapper.club/user-guide/controlling-mappings/#intent-api-230

**Comment by githubsterer** [comment]
I didn't know that. Cool!

**Comment by githubsterer** [comment]
For anyone who wants to know how to do this, I'm happy to explain, but I set up a Tasker intent to toggle Key Mapper that is called by Touch the Notch, and it's a thing of beauty!

The result is that I can pause Key Mapper by simply touching the front camera. It works awesome, and I would suggest that Seth implements this functionality inside the app if possible because it's extraordinary how well it works in my work flow.

**Comment by jambl3r** [comment]
You will be able to do this with floating buttons ;)
A small floating button placed behind the notch would provide similar functionality

**Comment by githubsterer** [comment]
Yes that's perfect. In the meantime as a proof of concept it works very well.

**Comment by jambl3r** [comment]
Quick tile, intent and notification give 3 alternatives to this which work better
